components/constellation/constellation.rs

```diff
@@ -735,8 +735,12 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
     }
 
     /// Create a new frame and update the internal bookkeeping.
-    fn new_frame(&mut self, frame_id: FrameId, pipeline_id: PipelineId, load_data: LoadData) {
-        let frame = Frame::new(frame_id, pipeline_id, load_data);
+    fn new_frame(&mut self,
+                 frame_id: FrameId,
+                 pipeline_id: PipelineId,
+                 load_data: LoadData,
+                 doc_type: DocumentType) {
```

Comment:
Still not sure why we need a `DocumentType` here. If we're creating a new frame, isn't it always the case that it starts off as an initial about:blank?